docs/source/architecture.rst

```diff
@@ -1,17 +1,142 @@
+a period of time, and `secondary nodes` monitor the health of the primary.
+The network switches to a new primary (called a `view change`) at regular
+intervals, as well as when the current primary is faulty.
+
+Sawtooth PBFT runs on each node in the network as a `consensus engine`, a
+separate process that handles consensus-related functionality and communicates
+with the validator through the consensus API.
+
+The following sections describe Sawtooth PBFT architecture:
+
+* :ref:`Network overview <network-overview-label>`: Describes PBFT fault
+  tolerance, view changes, sequence numbers, and the information stored
+  by each node
+
+* :ref:`Consensus messages <consensus-messages-label>`: Explains consensus
+  message structures and message types
+
+* :ref:`Sawtooth PBFT operation <pbft-operation-label>`: Shows how the
+  algorithm handles initialization, normal mode (block processing), and view
+  changes
+
+.. _network-overview-label:
+
+Network Overview
+================
+
+Fault Tolerance
+---------------
+
+A PBFT network consists of nodes that are ordered from 0 to `n-1`, where
+`n` is the total number of nodes in the network.
+
+The PBFT algorithm guarantees network `safety
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+as long as the number of faulty nodes remains below the required percentage.
+The maximum number of faulty nodes that the network can tolerate is determined
+by the formula :math:`f = \frac{n - 1}{3}`. In other words, no more than one
+third of the nodes (rounded down) can be out of order or dishonest at any given
+time.
+
+For example, a four-node network (the minimum required number of nodes)
+can tolerate one faulty node. Increasing the size of the network reduces the
+likelihood that that all :math:`\frac{n - 1}{3}` nodes would be faulty at the
+same time.
+
+Network `liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+is guaranteed as long as there are less than :math:`2f+1` unreachable nodes,
+counting both node and network failures. That is, at least third of the nodes
+(rounded up) must be reachable at all times.
+
+View Changes: Choosing a New Primary
+------------------------------------
+
+A PBFT network has one leader, called the `primary node`. All other nodes on the
+network are called `secondary nodes`.
+
+* The primary requests new blocks from the validator, notifies the network when
+  a new block is available, and reports consensus results (commit or fail) back
+  to the validator.
+
+* The secondary nodes monitor the health of the primary and request a change
+  if the primary appears out of order or malicious.
+
+A `view` is the period of time that a given node is the primary. The next
+primary is selected in a round-robin (circular) fashion, according to the order
+of nodes listed in the ``sawtooth.consensus.pbft.peers``
+:doc:`on-chain setting <on-chain-setting>`.
+In a four-node network, for example, node 0 is the primary at view 0,
```

Comment:
Would be helpful to specify, parenthetically, which node is node 0, node 1, etc. I.e. the 0th node in the list, the 1st node in the list, etc.